### PR TITLE
Refactored version

### PR DESCRIPTION
**Pull request by mstaniak**

### CODE REVIEW

File: R/MSstats_exports.R [hunk 1]
```diff
@@ -0,0 +1,564 @@
...
+    useUniquePeptide = TRUE, summaryforMultipleRows = max, 
+    fewMeasurements = "remove", removeMpeptides = FALSE,
+    removeOxidationMpeptides = FALSE, removeProtein_with1Peptide = FALSE, ...
+) {
+    input = MSstatsImport(list(evidence = evidence, protein_groups = proteinGroups), 
+                          type = "MSstats", tool = "MaxQuant", ...)
+    input = MSstatsClean(input, protein_id_col = proteinID, remove_by_site = TRUE)
+    annotation = .makeAnnotation(input, annotation, "Run" = "Rawfile")
+    
+    m_filter = list(col_name = "PeptideSequence", pattern = "M",
```
Comment (MeenaChoi) [inline R/MSstats_exports.R:72]: m_filter is not used in this function

Comment (mstaniak) [reply R/MSstats_exports.R:72]: It is on line 82, I think

File: R/MSstatsConvert_core_functions.R [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+#' @export
+#' 
+MSstatsImport = function(input_files, type, tool, tool_version = NULL, ...) {
+    checkmate::assertChoice(tool, 
+                            c("DIAUmpire", "MaxQuant", "OpenMS", "OpenSWATH",
+                              "Progenesis", "ProteomeDiscoverer", "Skyline",
+                              "SpectroMine", "Spectronaut"))
+    checkmate::assertChoice(type, c("MSstats", "MSstatsTMT"))
+    checkmate::assertTRUE(!is.null(names(input_files)))
+
```
Comment (huang704) [inline R/MSstatsConvert_core_functions.R:103]: Ting's comments

File: R/MSstats_exports.R [hunk 3]
```diff
@@ -0,0 +1,564 @@
...
+MaxQtoMSstatsTMTFormat = function(
+    evidence, proteinGroups, annotation, which.proteinid = 'Proteins',
+    rmProt_Only.identified.by.site = FALSE, useUniquePeptide = TRUE,
+    rmPSM_withMissing_withinRun = FALSE, rmPSM_withfewMea_withinRun = TRUE,
+    rmProtein_with1Feature = FALSE, summaryforMultipleRows = sum, ...
+) {
+    input = MSstatsImport(list(evidence = evidence, protein_groups = proteinGroups), 
+                          "MSstatsTMT", "MaxQuant", ...)
+    input = MSstatsClean(input, protein_id_col = which.proteinid, 
+                         remove_by_site = rmProt_Only.identified.by.site,
```
Comment (MeenaChoi) [inline R/MSstats_exports.R:120]: remove this option in MaxQtoMSstatsTMTformat. to make the consistent withLF

**Review by MeenaChoi — APPROVED** [review]

**Pull request merged**